toolchains/xslt-M4/document/json/object-reference-html.xsl

```diff
@@ -58,15 +58,28 @@
       <xsl:variable name="header-tag" select="if ($level le 6) then ('h' || $level) else 'p'"/>
       <xsl:variable name="grouped-object" select="(self::array | self::singleton-or-array)/*"/>
       <xsl:variable name="me" select="($grouped-object,.)[1]"/>
+
       <div class="model-entry definition { tokenize($me/@_metaschema-json-id,'/')[2] }">
          <xsl:variable name="header-class" expand-text="true">{ if (exists(parent::map)) then 'definition' else 'instance' }-header</xsl:variable>
          <div class="{ $header-class }">
-            <!-- generates h1-hx headers picked up by Hugo toc -->
-            <xsl:element expand-text="true" name="{ $header-tag }" namespace="http://www.w3.org/1999/xhtml">
-               <xsl:attribute name="id" select="@_tree-json-id"/>
-               <xsl:attribute name="class">toc{ $level} name</xsl:attribute>
-               <xsl:text>{ @key }</xsl:text>
+            <!-- ===!!! generates h1-hx headers picked up by Hugo toc !!!=== -->
+
+            <!-- ===!!! Anchor hard wrap-around of Headers 1-6 [see logic above with ($level le 6)] !!!=== -->               
+            <xsl:element expand-text="true" name="a" namespace="http://www.w3.org/1999/xhtml">
```

Comment:
It is possible to mix LRE (literal result element) and xsl:instructions for node generation here

```xml
<a xsl:expand-text="true" href="#{@_tree-json-id}">
   <xsl:attribute name="class">no-anchor-xslt Obj-Ref-Html--Xsl-71 toc{$level} name</xsl:attribute>
   <xsl:attribute name="title">Focus on {@key} details</xsl:attribute>
...
</a>
```

i.e. `<xsl:element>` is unnecessary if the name is known.

Unless you think the way you have it is more legible!

Reply:
This seems like a code style discussion. The original code used xsl:element so I see no problem keeping it.

Reply:
Is code style out of bounds? I offered this for consideration. AJ offers he likes it the other way which is fine. I just don't want things to slip because no one is watching. Personally I find the LRE syntax much more legible. (Except when it isn't, such as sometimes with attributes --  `xsl:attribute` is much easier to live with than `xsl:element`.)

Seems there are three ways we could approach this:

- take on code style / commenting / linting discussion head on (I would have to learn to follow the rules)
- take on code style discussion piecemeal (as here)
- avoid it for time being

I guess we are more or less in the second or third categories eh? Should I *not* pipe up when issues of code style come up (and expect similarly)? 😎 (Note that I'm opinionated only when it comes to XSLT and XQuery, with much to learn in any case.)